### DISCUSSION TITLE
Orders with multiple take profit price and size?

### DISCUSSION THREAD

**Discussion by djwatt3** [kernc/backtesting.py#991] [Q&A] [answered]
How I can place a new buy order with multiple take profits and different size?
example

```python
sizes = [0.50, 0.25, 0.25]
order = self.buy(sl=stopLoss)
order.setTP1(size=sizes[0], limit=tp1)
order.setTP2(size=sizes[1], limit=tp2)
order.setTP3(size=sizes[2], limit=tp3)
``` 

Thanks

**Comment by bonsai89** [comment]
Can this be done?

**Comment by djwatt3** [comment, marked as answer]
Yes,

```
trade = self.trades[0]
trade.close(portion = 0.50)
```

in this way you'll close 50% of your last order.

btw I switched to another framework better documented with a community and that allow you to do automatic live trading.

[https://github.com/freqtrade/freqtrade](url)

**Reply by bonsai89** [reply]
Thanks mate!
